docs/next/util/getRealFileBuffer.ts

```diff
@@ -0,0 +1,17 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+// Render files from the local content folder
+const ROOT = path.resolve('../content');
+
+/**
+ * For a given input path, validate that the requested file exists within the content folder.
+ * If it does, return the file buffer.
+ */
+export async function getRealFileBuffer(inputPath: string) {
+  const filePath = await fs.realpath(path.join(ROOT, inputPath));
+  if (!filePath.startsWith(ROOT)) {
+    throw new Error('File not found');
+  }
+  return await fs.readFile(filePath);
+}
```

Comment:
This is based on the recommendations in https://github.com/dagster-io/dagster/security/code-scanning/26.